src/main/java/de/zalando/aruha/nakadi/enrichment/Enrichment.java

```diff
@@ -0,0 +1,49 @@
+package de.zalando.aruha.nakadi.enrichment;
+
+import de.zalando.aruha.nakadi.domain.EnrichmentStrategyDescriptor;
+import de.zalando.aruha.nakadi.domain.EventCategory;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.exceptions.EnrichmentException;
+import de.zalando.aruha.nakadi.exceptions.InvalidEventTypeException;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Enrichment {
+    private final EnrichmentsRegistry registry;
+
+    public Enrichment(final EnrichmentsRegistry registry) {
+        this.registry = registry;
+    }
+
+    public void validate(final EventType eventType) throws InvalidEventTypeException {
+        if (eventType.getCategory() == EventCategory.UNDEFINED && !eventType.getEnrichmentStrategies().isEmpty()) {
+            throw new InvalidEventTypeException("must not define enrichment strategy for undefined event type");
+        }
+
+        final Set<EnrichmentStrategyDescriptor> uniqueStrategies = new HashSet<EnrichmentStrategyDescriptor>();
```

Comment:
1) Diamond <> can be used.
2) HashSet constructor can take collection as a parameter, so these two lines can be replaced with one.